event_sync/utils.go

```diff
@@ -0,0 +1,48 @@
+package event_sync
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+)
+
+func callAPIPost(path string, req, resp interface{}) error {
+	var body bytes.Buffer
+	err := json.NewEncoder(&body).Encode(&req)
+	if err != nil {
+		return err
+	}
+
+	response, err := http.Post(path, "application/json", &body)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return errors.New("POST request failed. Status: " + strconv.Itoa(response.StatusCode))
+	}
+
+	err = json.NewDecoder(response.Body).Decode(&resp)
```

Comment:
I would just directly return the call expression, rather than assigning to `err` and the `if` statement.